functions/go/set-image/image_transformer_test.go

```diff
@@ -0,0 +1,271 @@
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/fn/sdk"
+)
+
+// helper function to convert ResourceList items to yaml
+func itemsToYaml(items []*sdk.KubeObject) string {
```

Comment:
@mengqiy the new SDK exposes a `ToYaml` method for the `ResourceList` type, which includes the `FunctionConfig` in addition to the `Items`. For the purpose of testing expected output I figured it would be more simple/concise to exclude the `FunctionConfig` from the yaml and only include the `Items` (as is done in the unit tests for many of the other functions).

I was thinking perhaps we could push this logic up to the SDK by changing the type of `Items` from `[]*KubeObject` to a new type (e.g. `KubeObjects`) which could include its own `ToYaml` method. WDYT?